packages/product-feed-luigis-box/src/Model/Product/LuigisBoxProductRepository.php

```diff
@@ -31,8 +32,12 @@ public function getProducts(
         int $maxResults,
     ): iterable {
         $queryBuilder = $this->productRepository->getAllOfferedQueryBuilder($domainConfig->getId(), $pricingGroup)
-            ->addSelect('b')->leftJoin('p.brand', 'b')
-            ->orderBy('p.id', 'asc')
+            ->addSelect('b')
+            ->leftJoin('p.brand', 'b')
+            ->andWhere('p.variantType != :mainVariantType')
+            ->setParameter('mainVariantType', Product::VARIANT_TYPE_MAIN)
+            ->leftJoin('p.mainVariant', 'mv')
+            ->orderBy('mv.id', 'asc')
```

Comment:
Why is this necessary, please? 
Could not be the same result achieved without the join just by `->orderBy('IDENTITY(p.mainVariant)', 'asc')`? (I am not sure whether this would work :smile: )

Reply:
ah, I have found a similar usage in our code - https://github.com/shopsys/shopsys/blob/14.0/project-base/app/src/Model/Product/Parameter/ParameterRepository.php#L330